Cubical/Categories/Displayed/Section/Base.agda

```diff
@@ -0,0 +1,306 @@
+   definitional isomorphisms.
+
+   Style advice is to use Sections by default, and use Functorᴰ only
+   for things like displayed adjoints where they make more sense
+   intuitively.
+
+   Elimination rules are naturally stated in terms of local sections
+   (and often global sections, see below). Intro rules can be
+   formulated as either constructing a Section or a Functorᴰ. Good
+   style is to offer both intro for Section and intro' for Functorᴰ.
+
+-}
+{-# OPTIONS --safe #-}
+module Cubical.Categories.Displayed.Section.Base where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.HLevels
+
+open import Cubical.Data.Sigma
+import      Cubical.Data.Equality as Eq
+import      Cubical.Data.Equality.More as Eq
+
+open import Cubical.Categories.Category
+open import Cubical.Categories.Functor
+open import Cubical.Categories.Functors.Equality
+open import Cubical.Categories.Displayed.Base
+open import Cubical.Categories.Displayed.Functor
+import      Cubical.Categories.Displayed.Reasoning as HomᴰReasoning
+
+private
+  variable
+    ℓB ℓB' ℓBᴰ ℓBᴰ' ℓC ℓC' ℓCᴰ ℓCᴰ' ℓD ℓD' ℓDᴰ ℓDᴰ' ℓE ℓE' ℓEᴰ ℓEᴰ' : Level
+
+open Category
+open Categoryᴰ
+open Functor
+
+module _ {C : Category ℓC ℓC'}
+         {D : Category ℓD ℓD'}
+         (F : Functor D C)
+         (Cᴰ : Categoryᴰ C ℓCᴰ ℓCᴰ')
+         where
+  private
+    module C = Category C
+    module D = Category D
+    module Cᴰ = Categoryᴰ Cᴰ
+    module F = Functor F
+
+  -- Section without a qualifier means *local* section.
+  record Section : Type (ℓ-max (ℓ-max ℓC ℓC')
+                        (ℓ-max (ℓ-max ℓD ℓD')
+                        (ℓ-max ℓCᴰ ℓCᴰ')))
+    where
+    field
+      F-obᴰ  : ∀ d → Cᴰ.ob[ F ⟅ d ⟆ ]
+      F-homᴰ : ∀ {d d'} (f : D.Hom[ d , d' ])
+        → Cᴰ.Hom[ F ⟪ f ⟫ ][ F-obᴰ d , F-obᴰ d' ]
+      F-idᴰ : ∀ {d} → F-homᴰ (D.id {d}) Cᴰ.≡[ F .F-id ] Cᴰ.idᴰ
+      F-seqᴰ : ∀ {d d' d''}
+            → (f : D.Hom[ d , d' ])(g : D.Hom[ d' , d'' ])
+            → F-homᴰ (f D.⋆ g) Cᴰ.≡[ F .F-seq f g ] F-homᴰ f Cᴰ.⋆ᴰ F-homᴰ g
+
+{-
+   A Global Section is a local section over the identity functor.
+
+          A
+        ↗ |
+       /  |
+    M /   |
+     /    |
+    /     ↓
+   Γ ==== Γ
+
+
+   So global sections are by definition local sections
+
+   All local sections can be implemented as global sections into the
+   reindexed displayed category. See Reindex.agda for details.
+
+   But this isomorphism is not a definitional equality. Constructing a
+   local section is preferable as they are more flexible, but often
+   elimination principles are naturally proven first about global
+   sections and then generalized to local sections using reindexing.
+
+   Style is offer both: elim to construct a local section and elim' to
+   construct a global section.
+-}
+
+module _ {C : Category ℓC ℓC'}
+         (Cᴰ : Categoryᴰ C ℓCᴰ ℓCᴰ')
+         where
+  GlobalSection : Type _
+  GlobalSection = Section Id Cᴰ
```

Comment:
I may find the answer as I keep reading, but I still don't get the global v. local terminology here.

Reply:
A global section corresponds to $$\bar{x} : Γ ⊢ M(\bar{x}) : A(\bar{x})$$
while a local section corresponds to $$\bar{z} : Δ ⊢ M(\bar{z}) : A(γ(\bar{z}))$$ or equivalently $$\bar{z} : Δ ⊢ M(\bar{z}) : γ^*(A(\bar{x}))$$? In what sense is one global and the other local?

Reply:
I suppose it's just that the local section is only a section over "a subset" (very loosely construed) of A.
(Thinking of [fiber bundles](https://en.wikipedia.org/wiki/Section_(fiber_bundle))).

I guess that's it.